lib/bolt/transport/winrm/connection.rb

```diff
@@ -408,30 +315,21 @@ def execute(command)
           raise
         end
 
-        # 10 minutes in seconds
-        DEFAULT_EXECUTION_TIMEOUT = 10 * 60
-
-        def execute_process(path = '', arguments = [], stdin = nil,
-                            timeout = DEFAULT_EXECUTION_TIMEOUT)
+        def execute_process(path = '', arguments = [], stdin = nil)
           quoted_args = arguments.map do |arg|
             "'" + arg.gsub("'", "''") + "'"
-          end.join(',')
-
+          end.join(' ')
+
+          exec_cmd =
+            if stdin.nil?
+              "& #{path} #{quoted_args}"
+            else
+              "@'\n#{stdin}\n'@ | & #{path} #{quoted_args}"
```

Comment:
Unfortunately this isn't how `stdin` works on Windows. That's a PowerShell pipe operator, not a stdin redirection operator.

Reply:
I don't think I understand where that matters. This will always be a non-powershell script or task, and the examples I've tried all read happily read this from stdin.

Reply:
I'll give it a run through with .bat, .py, and .rb as some examples to try.

Reply:
This may be a problem for powershell tasks that use input_method=stdin. Need to explore that.

Reply:
Apparently this *will* redirect in some circumstances (but not to a PowerShell script).

For instance, given this batch file:

```batch
@echo off

ECHO Called %~nx0
ECHO From %~dp0
setlocal
for /F "tokens=*" %%a in ('more') do (
  echo #%%a
)
```

The output is

```
C:\Users\Administrator> 'foo' | C:\source\bolt\foo.bat
Called foo.bat
From C:\source\bolt\
#foo
```


There are some tricks you can use to work around PowerShell scripts treating that as pipeline input - https://stackoverflow.com/questions/38667584/getting-stdin-into-the-powershell-stream

Reply:
Also - I'm not sure that this method will work properly depending on system codepage, encoding of the passed string, etc. That would be worth investigating by calling a Python or Ruby script.

Reply:
I've added some additional testing with powershell scripts, they seem to read stdin happily via either `$Input` or `[Console]::In.ReadLine()`. I think the way we're invoking the script here results in a powershell subprocess that gets passed data on the pipe as stdin.

I'm not exactly sure how to test the codepage concerns you mention, and don't think we had any prior testing around that. I can mess around with UTF-8 to a Ruby script. Were you also imagining trying that with a different code page setup on the target?

Reply:
I think I've dealt with encoding now too.